src/cli/self_update.rs

```diff
@@ -1378,8 +1377,14 @@ fn get_new_rustup_version(path: &Path) -> Option<String> {
 }
 
 fn parse_new_rustup_version(version: String) -> String {
-    let re = Regex::new(r"\d+.\d+.\d+[0-9a-zA-Z-]*").unwrap();
-    let capture = re.captures(&version);
+    use lazy_static::lazy_static;
+    use regex::Regex;
+
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"\d+.\d+.\d+[0-9a-zA-Z-]*").unwrap();
+    }
+
+    let capture = RE.captures(&version);
```

Comment:
Have you confirmed this change is a performance win? There is some added complexity to the code (and lazy statics are not free - they use TLS and require taking a lock, I think), so this change could make no difference or even make things worse (as well as making the code a little more complicated).

Reply:
I think so, because regex docs seems to encourage this use of lazy_static: https://docs.rs/regex/1.1.5/regex/#example-avoid-compiling-the-same-regex-in-a-loop

Reply:
:+1: